docs/source/en/model_doc/audioflamingo3.md

```diff
@@ -0,0 +1,393 @@
+                },
+                {"type": "audio", "path": "https://audioflamingo3.github.io/static/chat/FPSbCAANfbJLVSwD.mp3"},
+            ],
+        }
+    ],
+]
+
+batch = processor.apply_chat_template(
+    conversations,
+    tokenize=True,
+    add_generation_prompt=True,
+    return_dict=True,
+).to(model.device)
+
+gen_ids = model.generate(**batch)
+
+inp_len = batch["input_ids"].shape[1]
+new_tokens = gen_ids[:, inp_len:]
+texts = processor.batch_decode(new_tokens, skip_special_tokens=True)
+print(texts)
+```
+
+➡️ transcription shortcut
+
+```python
+from transformers import AudioFlamingo3ForConditionalGeneration, AutoProcessor
+
+MODEL_ID = "nvidia/audio-flamingo-3-hf"
+processor = AutoProcessor.from_pretrained(MODEL_ID)
+model = AudioFlamingo3ForConditionalGeneration.from_pretrained(MODEL_ID, device_map="auto")
+
+inputs = processor.apply_transcription_request(audio="https://audioflamingo3.github.io/static/long_speech/t_837b89f2-26aa-4ee2-bdf6-f73f0dd59b26.wav").to(model.device)
+
+generated = model.generate(**inputs)
+transcription = processor.batch_decode(generated[:, inputs.input_ids.shape[1]:], skip_special_tokens=True)
+
+print(transcription[0])
+```
+
+## How the model works
+
+### Architecture
+
+* **AudioFlamingo3Encoder**
+  Whisper-style feature extractor + encoder → average-pool over time (stride 2) → LayerNorm.
+  Produces per-frame hidden states at the post-pool rate.
+
+* **AudioFlamingo3MultiModalProjector**
+  A small MLP that maps encoder features to the language model’s hidden size.
+
+* **AudioFlamingo3ForConditionalGeneration**
+  A causal language model that accepts text embeddings where each `<sound>` token slot is replaced, in place, by an audio frame embedding. No sequence-length change is introduced by fusion.
+
+### Processor-level alignment
+
+1. Each raw waveform is split into fixed-length windows based on the feature extractor’s `chunk_length` (seconds) and `sampling_rate` (Hz).
+2. For each window, the processor computes the number of post-pool frames `post_pool_len` that the encoder will output (matching the conv/pool schedule).
+3. The processor expands the audio placeholder token by the total number of post-pool frames across all windows.
+
+   * If the prompt contains no `<sound>`, all expanded tokens are inserted at the start of the user message in the chat template (or prepended to plain text if no template is used).
+   * If the prompt contains `<sound>`, exactly one placeholder is supported and it is expanded in place. Multiple placeholders are not supported.
+4. The model later replaces those token positions with the corresponding projected audio embeddings.
+
+This design guarantees a 1:1 match between placeholder positions in the text and encoder frame outputs, enabling safe batching without ragged audio–text concatenation.
+
+## Usage patterns
+
+### Transcription shortcut
```

Comment:
great, thanks for adding this example! 

@eustlb just want to double-check that this API looks good? We went for something similar to Voxtral: https://github.com/huggingface/transformers/blob/b9f90dc388fd415a2ba2a6a31a372f451d4a4eed/src/transformers/models/voxtral/processing_voxtral.py#L278